src/commands.rs

```diff
@@ -545,6 +545,44 @@ implement_commands! {
     }
 
     // list operations
+    
+    /// Pop an element from the front of a list, push it to the front of another list
+    /// and return it; or block until one is available
+    fn blmove_left_to_left<K: ToRedisArgs>(srckey: K, dstkey: K, timeout: usize) {
+        cmd("BLMOVE").arg(srckey).arg(dstkey).arg("LEFT").arg("LEFT").arg(timeout)
+    }
+
+    /// Pop an element from the front of a list, push it to the end of another list
+    /// and return it; or block until one is available
+    fn blmove_left_to_right<K: ToRedisArgs>(srckey: K, dstkey: K, timeout: usize) {
+        cmd("BLMOVE").arg(srckey).arg(dstkey).arg("LEFT").arg("RIGHT").arg(timeout)
+    }
+
+    /// Pop an element from the end of a list, push it to the front of another list
+    /// and return it; or block until one is available
+    fn blmove_right_to_left<K: ToRedisArgs>(srckey: K, dstkey: K, timeout: usize) {
+        cmd("BLMOVE").arg(srckey).arg(dstkey).arg("RIGHT").arg("LEFT").arg(timeout)
+    }
+
+    /// Pop an element from the end of a list, push it to the end of another list
+    /// and return it; or block until one is available
+    fn blmove_right_to_right<K: ToRedisArgs>(srckey: K, dstkey: K, timeout: usize) {
+        cmd("BLMOVE").arg(srckey).arg(dstkey).arg("RIGHT").arg("RIGHT").arg(timeout)
+    }
+
+    /// Pops `count` elements from the start of the first non-empty list key from the list of 
+    /// provided key names - defaults to 1 element if count not defined. Or blocks
+    /// until one is available.
+    fn blmpop_left<K: ToRedisArgs>(timeout: usize, numkeys: usize, key: K, count: Option<core::num::NonZeroUsize>) {
+        cmd("BLMPOP").arg(timeout).arg(numkeys).arg(key).arg("LEFT").arg("COUNT").arg(count)
+    }
+
+    /// Pops `count` elements from the end of the first non-empty list key from the list of 
+    /// provided key names - defaults to 1 element if count not defined. Or blocks
+    /// until one is available.
+    fn blmpop_right<K: ToRedisArgs>(timeout: usize, numkeys: usize, key: K, count: Option<core::num::NonZeroUsize>) {
+        cmd("BLMPOP").arg(timeout).arg(numkeys).arg(key).arg("RIGHT").arg("COUNT").arg(count)
+    }
```

Comment:
Not sure how arg is handling the `Option` here. If it is defaulting it to `1` then this should work but if it is converting it to just no arg then this will need tweaking as the new commands require the COUNT to be passed a number as the next arg.

Reply:
It will default to no argument on `None`. May be better to just pass a `usize`?